refactor: footer component

crnkovic opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure you're familiar with and follow the instructions in the [contributing guidelines](https://ark.dev/docs/program-incentives/guidelines/contributing).

Please fill out the information below to expedite the review and (hopefully) merge of your pull request!
-->

## Summary

https://app.clickup.com/t/2b006tr

Since most of our footers are pretty similar, this PR refactors the footer UI component to allow more "our classic" design. Below are few examples taken from our sites with code snippet and the screenshot of the footer that code would produce.

### Current usage

Current usage looks a bit wonky. This is one example:

```blade
<x-ark-footer
    :is-ark-product="false"
    :socials="[]"
    wrapper-class="flex max-w-screen-2xl"
>
    <x-slot name="copyText">
        <a href="{{ route('home') }}" class="transition-default hover:text-theme-secondary-200">
            @lang ('general.site_name')
        </a>

        <span>|</span>

        @lang('pages.footer.copyright')

        <a
            href="@lang('urls.ardent')"
            target="_blank"
            rel="noopener"
            class="underline hover:no-underline transition-default hover:text-theme-secondary-200"
        >@lang('general.ardent')</a>
    </x-slot>

    <x-slot name="copyrightSlot">
        <span class="break-words">
            <span>|</span>

            <span>@lang('pages.footer.all_rights_reserved')</span>

            <span>|</span>

            <a
                class="underline hover:no-underline transition-default hover:text-theme-secondary-200"
                href="{{ route('privacy-policy') }}"
            >@lang('general.privacy_policy')</a>

            <span>|</span>

            <a
                class="underline hover:no-underline transition-default hover:text-theme-secondary-200"
                href="{{ route('terms-of-service') }}"
            >@lang('general.terms_of_service')</a>
        </span>
    </x-slot>
</x-ark-footer>
```

... you can see it contains wonky slots, copyright text slots, etc. and this is pretty much repeated over the codebases.

### Usages

#### Without the name of the site

```blade
<x-ark-footer
    :creator="[
        'url' => '/',
        'label' => 'Ardent',
    ]"
    has-reserved-rights
    :policy="[
        'url' => '/privacy-policy',
        'label' => 'Privacy Policy',
    ]"
    :terms="[
        'url' => '/terms-of-service',
        'label' => 'Terms of Service',
    ]"
    :socials="[
        'twitter' => [
            'icon' => 'brands.solid.twitter',
            'url'  => trans('urls.socials.twitter'),
        ],
        'github' => [
            'icon' => 'brands.solid.github',
            'url'  => trans('urls.socials.github'),
        ],
    ]"
/>
```

![Screenshot 2022-08-03 at 15 24 06](https://user-images.githubusercontent.com/6536260/182618845-7cda78ea-69c7-4081-acde-3f92d3b98607.png)

#### With the name of the site

```blade
<x-ark-footer
    copyright-name="MSQ.io"
    copyright-url="/"
    :creator="[
        'url' => '/',
        'label' => 'Ardent',
    ]"
    has-reserved-rights
    :policy="[
        'url' => '/privacy-policy',
        'label' => 'Privacy Policy',
    ]"
    :terms="[
        'url' => '/terms-of-service',
        'label' => 'Terms & Conditions',
    ]"
    :socials="[
        //
    ]"
/>
```

![Screenshot 2022-08-03 at 15 27 27](https://user-images.githubusercontent.com/6536260/182619575-d4e6925d-6824-412c-b43d-ae02743a4fbc.png)

#### Without privacy policy / terms of service and "made with" link opens in new tab

```blade
<x-ark-footer
    :creator="[
        'url' => '/',
        'label' => 'ARK',
        'newWindow' => true,
    ]"
    has-reserved-rights
    :policy="[
        'url' => '/privacy-policy',
        'label' => trans('general.privacy_policy'),
    ]"
    :socials="[
        //
    ]"
/>
```

![Screenshot 2022-08-03 at 15 28 54](https://user-images.githubusercontent.com/6536260/182619890-e77e0bb0-b2a3-4305-89e8-80f67b954eba.png)

#### Simple

```blade
<x-ark-footer
    copyright-name="Site"
    copyright-url="/"
    has-reserved-rights
    :policy="[
        'url' => '/privacy-policy',
        'label' => trans('general.privacy_policy'),
    ]"
    :terms="[
        'url' => '/terms-of-service',
        'label' => 'Terms & Conditions',
    ]"
    :socials="[
        //
    ]"
/>
```

![Screenshot 2022-08-03 at 15 33 21](https://user-images.githubusercontent.com/6536260/182620909-5a921b94-dfbd-4ba1-a69a-eaee1b31b59a.png)

#### Appending custom content to the footer

```blade
<x-ark-footer
    :creator="[
        'url' => 'https://ardenthq.com',
        'label' => 'Ardent',
        'newWindow' => true,
    ]"
    has-reserved-rights
>
    Market Data by CoinGecko
</x-ark-footer>
```

![Screenshot 2022-08-04 at 16 11 36](https://user-images.githubusercontent.com/6536260/182868522-3236a650-b7ba-4123-ba4b-35dd30a72fef.png)

#### No props

```blade
<x-ark-footer />
```

![Screenshot 2022-08-04 at 16 12 40](https://user-images.githubusercontent.com/6536260/182868859-7b12a1dc-8270-47b2-9160-128c09d8c49a.png)

<!-- What changes are being made? -->

<!-- Why are these changes necessary? -->

<!-- How were these changes implemented? -->

## Checklist

<!-- Have you done all of these things (where applicable)?  -->

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

<!-- Feel free to add additional comments. -->

ItsANameToo commented on resources/views/footer.blade.php:
```diff
@@ -1,24 +1,100 @@
+    'name' => null,
+    'url' => null,
+    'creator' => [], // Needs `url` and `label` to display the "Made by"...
+    'arkProduct' => false,
```
```suggestion
    'isArkProduct' => false,
```

to make clear from the name it's a boolean

ItsANameToo commented on resources/views/footer.blade.php:
```diff
@@ -1,24 +1,100 @@
+    'url' => null,
+    'creator' => [], // Needs `url` and `label` to display the "Made by"...
+    'arkProduct' => false,
+    'reservedRights' => true, // Show "All rights reserved"...
```
```suggestion
    'hasReservedRights' => true, // Show "All rights reserved"...
```

ItsANameToo commented on resources/views/footer.blade.php:
```diff
@@ -1,24 +1,100 @@
+    'wrapperClass'  => 'px-8 max-w-7xl md:px-10',
+    'backgroundColor' => 'bg-theme-secondary-900',
+    'linkClass' => 'underline hover:no-underline transition-default hover:text-theme-secondary-200',
+    'name' => null,
+    'url' => null,
```
```suggestion
    'copyrightName' => null,
    'copyrightUrl' => null,
```

maybe? since they are only used there from what I saw

ItsANameToo commented on resources/views/footer.blade.php:
```diff
@@ -1,24 +1,100 @@
+            @if ((is_string($creator) || isset($creator['label'])) && $name)
+                <span class="mx-0.5">|</span>
+                <span>
+                    @lang ('ui::pages.footer.made_with_love') <a href="{{ $creator['url'] ?? url('/') }}" class="{{ $linkClass }}" target="_blank" rel="noopener">{{ is_string($creator) ? $creator : $creator['label'] }}</a>
```
this has `target="_blank"` while we open urls in the same window when they point to the existing domain (e.g. ark.io links to ark.io as "made with love by" thing and does not open in a new tab when clicked, similar for ardenthq). 

basically it should depend on the URL whether it uses `_blank` or not

ItsANameToo reviewed (changes requested):
- does this footer change cater for specifics like arkscan has with the marketdata addition:
<img width="531" alt="image" src="https://user-images.githubusercontent.com/35610748/182850425-b5b03370-e6e9-4d7b-aa9e-765b0a8fedc2.png">

- does the default `x-ark-footer` still work? as the `layout.blade.php` template makes use of that in foundation 🤔 
- will be a breaking change with the removal of the old footer component, so make sure that it works on all projects (added them to the clickup card as subtasks)

crnkovic commented:
@ItsANameToo good catch for ARKScan, added the ability to pass slot as custom content that's appended to the footer. Also added the ability to indicate whether to open links in new window or not. Will leave this as draft until I recreate footers from all projects.

Updated PR description with new stuff and new examples.
